packages/shared_preferences/shared_preferences_platform_interface/lib/method_channel_shared_preferences.dart

```diff
@@ -18,39 +18,29 @@ const MethodChannel _kChannel =
 class MethodChannelSharedPreferencesStore
     extends SharedPreferencesStorePlatform {
   @override
-  Future<bool> remove(String key) {
-    return _invokeBoolMethod('remove', <String, dynamic>{
-      'key': key,
-    });
+  Future<bool> remove(String key) async {
+    return await _kChannel.invokeMethod<bool>(
+      'remove',
+      <String, dynamic>{'key': key},
+    ) as bool;
   }
 
   @override
-  Future<bool> setValue(String valueType, String key, Object value) {
-    return _invokeBoolMethod('set$valueType', <String, dynamic>{
-      'key': key,
-      'value': value,
-    });
-  }
-
-  Future<bool> _invokeBoolMethod(String method, Map<String, dynamic> params) {
-    return _kChannel
-        .invokeMethod<bool>(method, params)
-        // TODO(yjbanov): I copied this from the original
-        //                shared_preferences.dart implementation, but I
-        //                actually do not know why it's necessary to pipe the
-        //                result through an identity function.
-        //
-        //                Source: https://github.com/flutter/plugins/blob/3a87296a40a2624d200917d58f036baa9fb18df8/packages/shared_preferences/lib/shared_preferences.dart#L134
-        .then<bool>((dynamic result) => result);
+  Future<bool> setValue(String valueType, String key, Object value) async {
+    return await _kChannel.invokeMethod<bool>(
+      'set$valueType',
+      <String, dynamic>{'key': key, 'value': value},
+    ) as bool;
   }
 
   @override
-  Future<bool> clear() {
-    return _kChannel.invokeMethod<bool>('clear');
+  Future<bool> clear() async {
+    return await _kChannel.invokeMethod<bool>('clear') as bool;
   }
 
   @override
-  Future<Map<String, Object>> getAll() {
-    return _kChannel.invokeMapMethod<String, Object>('getAll');
+  Future<Map<String, Object>> getAll() async {
+    return await _kChannel.invokeMapMethod<String, Object>('getAll')
+        as Map<String, Object>;
```

Comment:
Why do we need the cast? I thought the purpose of `invokeMapMethod` was do to that cast internally.

Reply:
All the invoke methods return the nullable version of the type parameters. As in `invokeMapMethod<String, Object>` returns `Map<String, Object>?`.

Reply:
Wouldn't that just need a `!` though, rather than a cast?

Reply:
Yes that would work but you would have to add parentheses around the await keyword. Basically the two options are

```dart
final Map<String, Object> map = (await invokeMapMethod<String, Object>())!;
```

or 

```dart
final Map<String, Object> map = await invokeMapMethod<String, Object>() as Map<String, Object>;
```

I don't really have a preference on which is better. I think the top one is shorter, but the second is easier to read/cleaner.

Reply:
I'd vote strongly for the top one, for the same reason that people prefer C++-style casts to C-style casts: it's explicit about what it's doing. `!` does exactly one thing (remove the nullability from the type), while `as` could do other kinds of casting. The fact that I didn't know what the cast was for in the second form seems like pretty good evidence that it has readability issues.

Plus, if there were a mistake with the types here (say, a mismatch in the contents of the `<>` in invokeMapMethod vs the cast), the `as` would be a runtime failure, while the `!` would be a compile-time failure.